prospector/filter_rank/ranker.py

```diff
@@ -41,4 +44,51 @@ def train(model_name: str, data_filename=TRAINING_DATA) -> str:
     # fit(dataframe) --> model
     # compute metrics
     # save model to disk (file: model_name)
-    return save_model("", model_name)
+    df = _get_training_dataframe(data_filename, num_elem_training_data)
+    if df is None:
+        return ""  # Special case, when returned dataframe is None
+
+    model = ""  # here should go the actual classifier model
+    # model.fit(df)
+    # compute metrics
+    return save_model(model, model_name)
+
+
+def _get_training_dataframe(data_filename=TRAINING_DATA, num_elem_training_data=NUM_ELEMENTS_TRAINING_DATA):
+    """
+    This is the helper function to construct the pandas dataframe object for the training data.
+    It returns pandas dataframe if succeeds and None otherwise
+    """
+    commits = []
+    try:
+        with open(data_filename, 'r', encoding='utf8') as f_in:
+            count = 0
+            for line in f_in:
+                line = line.strip()
+                line = re.sub(r"[\['\] ]", "", line).split(",")
```

Comment:
Sorry I forced you to do this :-)
We can have a proper CSV for the next iteration, so this code can be simplified